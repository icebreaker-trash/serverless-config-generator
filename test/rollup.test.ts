import { rollup } from 'rollup'
import omit from 'lodash/omit'
import type { OutputChunk, OutputAsset } from 'rollup'
import config from '../rollup.config'

function normalizeOutput(
  outputs: [OutputChunk, ...(OutputChunk | OutputAsset)[]]
) {
  return outputs.map((x) => {
    return omit(x, ['modules', 'facadeModuleId', 'moduleIds'])
  })
}
// diff output changes
// run `jest -u` to update snap
describe.skip('rollup build', () => {
  it('lib build output diff', async () => {
    // const result:RollupBuild[] = []

    const bundle = await rollup({
      input: config.input,
      external: config.external,
      output: config.output,
      plugins: config.plugins
    })
    if (Array.isArray(config.output)) {
      for (let j = 0; j < config.output.length; j++) {
        const { output } = await bundle.generate(config.output[j])
        expect(normalizeOutput(output)).toMatchSnapshot()
      }
    } else if (config.output) {
      const { output } = await bundle.generate(config.output)
      expect(normalizeOutput(output)).toMatchSnapshot()
    }
  })
})
