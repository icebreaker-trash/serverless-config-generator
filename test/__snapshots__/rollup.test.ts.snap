// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup build > lib build output diff 1`] = `
[
  {
    "code": "'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const hello = () => 'world';
console.log('Load util');

const foo = 'bar';
const wait = (timeout) => {
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve(true);
        }, timeout);
    });
};

exports.foo = foo;
exports.hello = hello;
exports.wait = wait;
",
    "dynamicImports": [],
    "exports": [
      "foo",
      "hello",
      "wait",
    ],
    "fileName": "index.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "name": "index",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;

exports[`rollup build > lib build output diff 2`] = `
[
  {
    "code": "const hello = () => 'world';
console.log('Load util');

const foo = 'bar';
const wait = (timeout) => {
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve(true);
        }, timeout);
    });
};

export { foo, hello, wait };
",
    "dynamicImports": [],
    "exports": [
      "foo",
      "hello",
      "wait",
    ],
    "fileName": "index.mjs",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "name": "index",
    "referencedFiles": [],
    "type": "chunk",
  },
]
`;
